File: quickstart/MyStrategy.cpp
#include "MyStrategy.hpp"
#include <exception>

MyStrategy::MyStrategy(const model::Constants &constants) {}

model::Order MyStrategy::getOrder(const model::Game &game, DebugInterface *debugInterface)
{
    std::unordered_map<int, model::UnitOrder> actions;
    for (auto &unit : game.units)
    {
        if (unit.playerId != game.myId)
            continue;

        std::shared_ptr<model::ActionOrder::Aim> aim = std::make_shared<model::ActionOrder::Aim>(true);
        std::optional<std::shared_ptr<model::ActionOrder>> action = std::make_optional(aim);

        model::UnitOrder order({-unit.position.x, -unit.position.y}, {-unit.direction.y, unit.direction.x}, action);
        actions.insert({unit.id, order});
    }
    return model::Order(actions);
}

void MyStrategy::debugUpdate(DebugInterface &debugInterface) {}

void MyStrategy::finish() {}

File: quickstart/MyStrategy.php
<?php

use Model\Constants;
use Model\Game;
use Model\Order;
use Model\Unit;
use Model\UnitOrder;
use Model\Vec2;

require_once 'Model/Game.php';
require_once 'Model/Order.php';
require_once 'Model/Constants.php';

class MyStrategy
{
    function __construct(Constants $constants)
    {
    }

    function getOrder(Game $game, ?DebugInterface $debugInterface): Order
    {
        $order = [];

        /** @var Unit[] $units */
        $units = $game->units;

        foreach ($units as $unit) {
            if ($unit->playerId != $game->myId) {
                continue;
            }

            $order[] = new UnitOrder(
                new Vec2(-$unit->position->x, -$unit->position->y),
                new Vec2(-$unit->direction->x, $unit->direction->y),
                new Model\ActionOrder\Aim(true)
            );
        }
        return new Order($order);
    }

    function debugUpdate(DebugInterface $debug_interface)
    {
    }

    function finish()
    {
    }
}